packages/x-date-pickers/src/internals/hooks/useField/useField.interfaces.ts

```diff
@@ -84,7 +84,37 @@ export interface FieldSection {
   separator: string | null;
   dateSectionName: MuiDateSectionName;
   formatValue: string;
-  query: string | null;
+  edited: boolean;
+}
+
+/**
+ * Object used to access and update the active value.
+ * Mainly useful in the range fields where we need to update the date containing the active section without impacting the other one.
+ */
+interface FieldActiveDateManager<TValue, TDate> {
+  /**
+   * Date containing the current active section.
```

Comment:
Out of the scop of this PR:
It let me think that this will be problematic for copying a full date range, because the selection will be contained in both dates

Reply:
Right now Ctrl A would select both dates
But not sure how to support single date copy / paste

As a user, I could expect Shift + Arrow Right to work to select several sections
Or maybe the double click (work on a text input but not on a date input in the html spec)